planner/core/find_best_task.go

```diff
@@ -418,6 +442,204 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 	return
 }
 
+func (ds *DataSource) convertToIndexMergeScan(prop *property.PhysicalProperty, candidate *candidatePath) (task task, err error) {
+	if prop.TaskTp != property.RootTaskType || !prop.IsEmpty() {
+		return invalidTask, nil
+	}
+	path := candidate.path
+	var totalCost float64
+	var totalRowCount float64
+	scans := make([]PhysicalPlan, 0, len(path.partialIndexPaths))
+	cop := &copTask{}
+	for _, partPath := range path.partialIndexPaths {
+		if partPath.isTablePath {
+			scan, partialCost, rowCount := ds.convertToPartialTableScan(prop, partPath)
+			scans = append(scans, scan)
+			totalCost += partialCost
+			totalRowCount += rowCount
+			continue
+		}
+		scan, partialCost, rowCount := ds.convertToPartialIndexScan(prop, partPath)
+		scans = append(scans, scan)
+		totalCost += partialCost
+		totalRowCount += rowCount
+	}
+	ts, partialCost := ds.convertToTableScanForIndexMerge(prop, path, totalRowCount)
+	totalCost += partialCost
+	cop.partPathForIdxMerge = scans
+	cop.tablePlan = ts
+	cop.cst = totalCost
+	task = cop
+	if prop.TaskTp == property.RootTaskType {
+		task = finishCopTask(ds.ctx, task)
+	} else if _, ok := task.(*rootTask); ok {
+		return invalidTask, nil
+	}
```

Comment:
We don't need this check anymore, since we have checked `prop.TaskTp` at the beginning of this function.